core/src/Kokkos_Core.hpp

```diff
@@ -202,7 +203,12 @@ class KOKKOS_ATTRIBUTE_NODISCARD ScopeGuard {
 #endif
   ScopeGuard(int& argc, char* argv[]) {
     sg_init = false;
-    if (!Kokkos::is_initialized()) {
+    if (!is_initialized()) {
+      if (is_finalized()) {
+        Kokkos::abort(
+            "Attempt to create a Kokkos::ScopeGuard after Kokkos::finalize was "
+            "called.");
+      }
```

Comment:
```suggestion
    if (is_finalized()) {
      Kokkos::abort(
          "Attempt to create a Kokkos::ScopeGuard after Kokkos::finalize was "
          "called.");
    }
    if (!is_initialized()) {
```

Reply:
it is one more check in the case of `is_initialized()==true` but I guess thats not that much cost?